### PR TITLE
Calling Reader SC for collecting subscriptions

### PR DESCRIPTION
**Pull request by ebarakos**
Calling SC methods of https://github.com/ritual-net/infernet-sdk/blob/8e6cd6f5cbd66dc9baacb895a2ed8fe2c9ee3b6f/src/utility/Reader.sol to reduce RPC calls on subscriptions collection

### CODE REVIEW

File: src/chain/listener.py [hunk 1]
```diff
@@ -109,61 +114,128 @@ def __init__(
...
+                        # If filtered out by guardian, message is irrelevant
+                        log.info("Ignored subscription creation", id=sub_id, err=filtered.error)
+                    else:
+                        # Pass filtered message to ChainProcessor
+                        create_task(self._processor.track(msg))
+                        log.info("Relayed subscription creation", id=sub_id)
+                    break
+        return
+    
+    # DEPRECATED, using _sync_batch_subscriptions_creation through Reader SC
```
Comment (ebarakos) [inline src/chain/listener.py:185]: Commented code should be removed, left it temporarily for potential rollback

File: config.sample.json [hunk 2]
```diff
@@ -25,7 +25,7 @@
     },
     "snapshot_sync": {
       "sleep": 1.5,
-      "batch_size": 200
+      "batch_size": 9653
```
Comment (evangelos-ritual) [inline config.sample.json:28]: Max number before getting SC error

Comment (ebarakos) [inline config.sample.json:28]: Max number before getting SC error

File: config.sample.json [hunk 3]
```diff
@@ -25,7 +25,7 @@
     },
     "snapshot_sync": {
       "sleep": 1.5,
-      "batch_size": 200
+      "batch_size": 1800
```
Comment (ebarakos) [inline config.sample.json:28]: Sane default as it is working with 5 public RPCs

**Pull request closed without merging**